rotkehlchen/chain/ethereum/modules/aave/graph.py

```diff
@@ -565,6 +639,65 @@ def _process_events(
             total_earned_liquidations=total_earned_liquidations,
         )
 
+    def _process_graph_query_result(
+        self,
+        query: Dict[str, Any],
+        deposits: List[AaveDepositWithdrawalEvent],
+        withdrawals: List[AaveDepositWithdrawalEvent],
+        borrows: List[AaveBorrowEvent],
+        repays: List[AaveRepayEvent],
+        liquidation_calls: List[AaveLiquidationEvent],
+        user_merged_data: Dict[str, Any],
+        from_ts: Timestamp,
+        to_ts: Timestamp,
+    ) -> None:
+        """
+        Given a query result from the graph this function extracts information for:
+        - deposits
+        - withdrawals
+        - borrows
+        - repays
+        - liquidation_calls
+        and extends the corresponding arguments with the obtained information.
+        """
+        if 'users' not in query or len(query['users']) == 0:
+            # If there is no information on the query finish the execution
+            log.debug(f'Aave subgraph query has no information for user. {str(query)}')
+            return
+
+        user_result = query['users'][0]
+        msg = 'Failed to obtain a valid result from Aave graph.'
+        try:
+            deposits += self._parse_deposits(user_result['depositHistory'], from_ts, to_ts)
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+        try:
+            withdrawals += self._parse_withdrawals(
+                withdrawals=user_result['redeemUnderlyingHistory'],
+                from_ts=from_ts,
+                to_ts=to_ts,
+            )
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+        try:
+            borrows += self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+        try:
+            repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
+        try:
+            liquidation_calls += self._parse_liquidations(
+                liquidations=user_result['liquidationCallHistory'],
+                from_ts=from_ts,
+                to_ts=to_ts,
+            )
+        except KeyError as e:
+            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
```

Comment:
Hmm ... why do so many try/catch?

```suggestion
        try:
            deposits += self._parse_deposits(user_result['depositHistory'], from_ts, to_ts)
            withdrawals += self._parse_withdrawals(
                withdrawals=user_result['redeemUnderlyingHistory'],
                from_ts=from_ts,
                to_ts=to_ts,
            )
            borrows += self._parse_borrows(user_result['borrowHistory'], from_ts, to_ts)
            repays += self._parse_repays(user_result['repayHistory'], from_ts, to_ts)
            liquidation_calls += self._parse_liquidations(
                liquidations=user_result['liquidationCallHistory'],
                from_ts=from_ts,
                to_ts=to_ts,
            )
        except KeyError as e:
            self.msg_aggregator.add_warning(msg + f' Missing key {str(e)}')
            return
```

Also .. `return` at the error handling. If a key is missing this is bad, means that the interface has changed so this entry is not to be trusted.

Reply:
I modified it from a simple try/catch to this multiple because if one key is missing but the others are present the job could continue. Now that I see it maybe doesn't make sense. In my head yesterday was a good idea

Reply:
I understand the intention but as you said, definitely does not make sense. Please change it.